src/components/Microsoft.DotNet.UpgradeAssistant.Analysis/Microsoft.DotNet.UpgradeAssistant.Analysis.csproj

```diff
@@ -0,0 +1,13 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net5.0</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\..\common\Microsoft.DotNet.UpgradeAssistant.Abstractions\Microsoft.DotNet.UpgradeAssistant.Abstractions.csproj" />
+    <ProjectReference Include="..\..\steps\Microsoft.DotNet.UpgradeAssistant.Steps.Packages\Microsoft.DotNet.UpgradeAssistant.Steps.Packages.csproj" />
```

Comment:
Should not reference any steps directly (they're technically in an extension). Anything provided by those steps should be resolved via the abstractions

Reply:
Yup - that makes the most sense. Each "extension" should register it's own "IAnalyzeResultProvider" where it makes sense

Reply:
@mjrousos By packages class, do you mean Microsoft.DotNet.UpgradeAssistant.Steps.Packages ? Then it would create a circular dependency since Analysis component project depends on the steps projects already and now steps will need to depend on analysis to DI IAnalyzeResultProvider.

Reply:
The packaging class can (and does) depend on the abstractions class. Any abstraction that you need shared should put in there. The analysis project should be the part that does the heavy lifting, but the interface that extensions implement should probably be in the abstractions project.